src/charts/line.js

```diff
@@ -510,15 +510,24 @@ define(function(require){
                 return d;
             });
 
-            // Normalize dataByTopic
-            dataByTopic.forEach(function(kv) {
-                kv.dates.forEach(function(d) {
-                    d.date = new Date(d[dateLabel]);
-                    d.value = +d[valueLabel];
-                });
-            });
+            const newDataByTopic = dataByTopic.reduce((accum, topic) => {
+                let {dates, ...restProps} = topic;
+
+                let newDates = dates.map(d => ({
+                       date: new Date(d[dateLabel]),
+                       value: +d[valueLabel],
+                       [dateLabel]: d[dateLabel]
```

Comment:
I guess we don't want this to be here.
In the cleaned data there should be only one 'date' entry, labeled as 'date'.